daft/io/catalog.py

```diff
@@ -34,15 +34,6 @@ class DataCatalogTable:
     table_name: str
     catalog_id: Optional[str] = None
 
-    def __post_init__(self) -> None:
-        import warnings
-
-        warnings.warn(
-            "This API is deprecated in daft >=0.5.0 and will be removed in >=0.6.0. Users should use the new functionality in daft.catalog.",
-            DeprecationWarning,
-            stacklevel=2,
-        )
-
```

Comment:
TBH I feel we could remote daft/io/catalog altogether. That can be in a different PR